### ISSUE TITLE
stable-1.9 branch's ec2.py has an unbound error variable

### ISSUE THREAD

**Issue by jmhodges** [ansible/ansible#10910]
This occurs in the stack trace below. Specifically, I consistently get it because a user I have does not have access to rds:DescribeDBInstances (which is a separate bug I might raise up).

```
~/ansible/plugins/inventory/ec2.py
Traceback (most recent call last):
  File "/home/jmhodges/ansible/plugins/inventory/ec2.py", line 802, in <module>
    Ec2Inventory()
  File "/home/jmhodges/ansible/plugins/inventory/ec2.py", line 156, in __init__
    self.do_api_calls_update_cache()
  File "/home/jmhodges/ansible/plugins/inventory/ec2.py", line 332, in do_api_calls_update_cache
    self.get_rds_instances_by_region(region)
  File "/home/jmhodges/ansible/plugins/inventory/ec2.py", line 389, in get_rds_instances_by_region
    self.fail_with_error(error)
UnboundLocalError: local variable 'error' referenced before assignment
```

**Comment by edrozenberg** [comment]
Is it related to https://github.com/ansible/ansible/issues/10840 ?

In my case I had to edit ec2.ini and set

```
rds = False
```

So that the ec2.py script won't try to do anything with rds. My account wasn't signed up to Amazon RDS and this was causing the problem.

**Comment by jmhodges** [comment]
That's another problem I don't intend for this bug to address.

This was is just that anisible crashes without useful error messages because the "error" variable is unbound in some paths.

**Comment by bcoca** [comment]
closing by merged #10985
